web/src/utils/iframes.ts

```diff
@@ -0,0 +1,190 @@
+import { TFunction } from 'i18next'
+
+import { CONSENT_ROUTE, ExternalSourcePermissions } from 'api-client'
+
+import { IFRAME_BLANK_SOURCE, IframeSources } from '../components/RemoteContent'
+import dimensions from '../constants/dimensions'
+
+export const EXTERNAL_SOURCES_COOKIE_NAME = 'Opt-In-External-Sources'
+export const addDoNotTrackParameter = (iframe: HTMLIFrameElement): void => {
+  if (iframe.src.includes('vimeo')) {
+    const url = new URL(iframe.src)
+    url.searchParams.append('dnt', '1')
+    iframe.setAttribute('src', url.href)
+  }
+}
+
+export const preserveIFrameSourcesFromContent = (
+  index: number,
+  source: string,
+  setExistingIframes: (sources: IframeSources) => void,
+  contentIframeSources: IframeSources,
+): void => {
+  const updatedContentSources: IframeSources = contentIframeSources
+  updatedContentSources[index] = source
+  setExistingIframes(updatedContentSources)
+}
+
+export const hideIframe = (iframe: HTMLIFrameElement): void => {
+  iframe.setAttribute('src', IFRAME_BLANK_SOURCE)
+  iframe.setAttribute('style', 'display:none')
+}
+
+export const restoreIframe = (iframe: HTMLIFrameElement, source: string): void => {
+  iframe.setAttribute('src', source)
+  iframe.setAttribute('style', 'display:block')
+}
```

Comment:
Optional and maybe not a good practice :smile::
Could we perhaps "save" the src in some of the iframes attributes, e.g. the id to avoid having to save it together with some index and update logic in some unrelated react state. At least I found the code very confusing/hard to read. I tried the following and it would work afaics:
```suggestion
export const hideIframe = (iframe: HTMLIFrameElement): void => {
  const source = iframe.getAttribute('src')
  if (source !== null && source !== IFRAME_BLANK_SOURCE) {
    iframe.setAttribute('id', source)
    iframe.setAttribute('src', IFRAME_BLANK_SOURCE)
    iframe.setAttribute('style', 'display:none')
  }
}

export const restoreIframe = (iframe: HTMLIFrameElement): void => {
  const source = iframe.getAttribute('id')
  if (source) {
    iframe.setAttribute('src', source)
    iframe.setAttribute('style', 'display:block')
  }
}
```
Just an idea though, so feel free to just ignore :)

Edit: Actually, it seems like you are already using `iframe-container${supportedSource}${iframeIndex}` as id for the container, so maybe that can easily be combined?

Reply:
i don't feel like its getting cleaner with this solution tbh